teerex/src/lib.rs

```diff
@@ -446,6 +500,78 @@ impl<T: Config> Pallet<T> {
 			Err(<Error<T>>::RemoteAttestationTooOld.into())
 		}
 	}
+
+	fn check_block_and_get_latest_header(
```

Comment:
Clean Code: A function containing an `and` in its name should be split ;)
I'm not even sure if this function is needed.. And if it's needed, the check only which could return a bool or a result.
The function `confirm_sidechain_block` can then be called after a successful check. No hidden function calls and no extra variable needed that don't make sense for "check" only (as the function name suggests)

Reply:
I inlined the function